src/lib/openapi-to-json-schema.ts

```diff
@@ -367,6 +362,35 @@ export default function toJSONSchema(
       }
     }
 
+    // To ease some of the burden on our frontend having to juggle mixed types we're opting to
+    // transform them into a `oneOf`.
+    if ('type' in schema && Array.isArray(schema.type)) {
+      schema.type = Array.from(new Set(schema.type));
+
+      // If we have a `null` type but there's only two types present then we can remove `null` as
+      // an option and flag the whole schema as `nullable`.
+      if (schema.type.includes('null')) {
+        schema.type = schema.type.filter(t => t !== 'null');
+        schema.nullable = true;
+      }
+
+      if (schema.type.length === 1) {
+        schema.type = schema.type.shift();
+      } else {
```

Comment:
wait so this block just makes the `schema.type` array empty, right? why are we doing this?

(not disagreeing, just not following)